### PR TITLE
fix(sync): replace MAX_PAGES=100 with cursor-validity guards (#679)

### LINKED ISSUE #679: Phoenix Portal sync fails at 100-page limit for large datasets
## Summary

A Phoenix Portal sync initiated from the mobile app cannot complete for a linked account whose pull requires more than 100 pages. The app processes exactly 100 pages / 10,007 entities, then terminates the pull with its hard maximum-page guard.

## Reporter

Discord: `iaminthedetails` (display name: Devil)

## Type / target

- Type: Bug
- Target: Project Phoenix mobile app (`project-phoenix-mp`)
- Surface: **Settings → Phoenix Portal**
- Platform: Android (inferred from the system navigation shown in the screenshots)
- App version / exact device: Not provided

## Steps to reproduce

1. In the mobile app, link a Premium Phoenix Portal account that has enough cloud data to require more than 100 pull pages.
2. Open **Settings → Phoenix Portal**.
3. Start a sync / full resync.
4. Allow the pull to continue through 100 pages.

## Expected behavior

The app should synchronize the complete account dataset, including datasets larger than 100 pages. If a safety bound is required, the sync should use a resumable/batched strategy rather than leaving the account only partially synchronized.

## Actual behavior

The progress UI advances normally—for example:

> Syncing... (10 pages, 1007 items)

After page 100, the app reports:

> Sync partially complete — pull failed: Pull exceeded maximum page limit (100). Processed 10007 entities across 100 pages. This may indicate a data issue - please contact support.

The account remains linked and the user is offered **Sync Now** / **Force Full Resync**, but the pull cannot pass the same hard limit.

## Impact

Accounts with large datasets cannot complete cross-device synchronization. Retrying is expected to hit the same deterministic page limit, so the failure is not recoverable from the current UI.

## Screenshot evidence

- [Progress at 10 pages / 1,007 items](https://cdn.discordapp.com/attachments/1509574296807407717/1530368267120803921/AP1GczNKj7A_JPQEKdbEJe96dmGTUeD7gFkCp0ZtvubwMZIRr0klqQCu53nvvAw389-h869-s-no-gm.png?ex=6a655203&is=6a640083&hm=d0010089768de9bdac7215c6ca064675b29291684168701c9283b6525c99ef27&)
- [Failure after 100 pages / 10,007 entities](https://cdn.discordapp.com/attachments/1509574296807407717/1530368267800154333/AP1GczO6g3x2_DUYdHnwEjX4eOp7aHu_RUP6oeNARQTzC2jdjiL7EwLFXx1cWgw389-h869-s-no-gm.png?ex=6a655203&is=6a640083&hm=1dc506c5b44c27c09d49ebc9170126c99d51812aab9d2db3388ca1369e72b9e5&)

The screenshots show a Premium linked account; its email address is intentionally omitted here because it is not needed to reproduce or diagnose the bug.

## Initial source-grounded triage

Current `main` defines `SyncConfig.MAX_PAGES = 100` and checks `pagesProcessed >= SyncConfig.MAX_PAGES` before fetching the next page in `SyncManager.pullRemoteChangesWithResult`. The resulting exception text is an exact match for the reporter's screenshot. With `DEFAULT_PAGE_SIZE = 100`, the observed 10,007 entities across 100 pages is consistent with a legitimate dataset reaching the fixed client guard.

This is related to, but not a duplicate of, #634 / PR #637. That change added per-page rate limiting and resumable handling of 429/503 responses; the current failure is the separate 100-page ceiling that remains after those changes.

## Original Discord report

```text
Issues syncing large datasets to web portal within app.  Cycles through 100 pages of syncing, then error is received.
```

### PR DESCRIPTION
**Pull request by 9thLevelSoftware**
## Summary

Replace the hard 100-page ceiling in `SyncManager.pullRemoteChangesWithResult` with pagination-protocol safety checks, fixing sync failures for accounts with >10,000 entities.

Fixes #679

## Problem

`SyncConfig.MAX_PAGES = 100` was a client-side-only guard that terminated pulls after 100 pages (100 x 100 entities = 10,000 entity cap). Accounts exceeding this limit received:

> Pull exceeded maximum page limit (100). Processed 10007 entities across 100 pages.

Retrying restarted from page 1 and hit the same deterministic ceiling, making the failure unrecoverable.

## Root Cause

The 100-page cap treated a valid, advancing cursor for a large account as a corrupt pagination loop. The portal's `mobile-sync-pull` edge function has no page limit and supports unlimited cursor-based pagination.

## Fix

- **Track seen continuation cursors** in a `MutableSet<String>`. A repeated cursor = server infinite loop -> fail with diagnostic error.
- **Reject blank continuation cursors** as protocol violations.
- **Raise `MAX_PAGES`** from 100 to 10,000 as a documented emergency circuit breaker (telemetry-backed, not a product limit).
- **Preserve all existing guards**: empty-page, null-cursor, rate-limiter (#637), retry handling, personalRecordIds omission.

## Changes

| File | Change |
|------|--------|
| `SyncManager.kt` | Replace `pagesProcessed >= MAX_PAGES` guard with cursor-validity checks; raise `MAX_PAGES` to 10,000 |
| `PortalPullPaginationTest.kt` | Add 4 tests: 101-page success, repeated cursor, blank cursor, null cursor |

## Tests Added

1. **`pullBeyond100PagesSucceedsWhenCursorsAreDistinct`**: 101 pages with distinct cursors -> full success
2. **`pullWithRepeatedCursorFailsWithDiagnosticError`**: repeated cursor -> failure with "repeated cursor" message, lastSync unchanged
3. **`pullWithBlankCursorFailsWithDiagnosticError`**: blank cursor -> failure with "blank continuation cursor" message, lastSync unchanged
4. **`pullWithNullCursorAndHasMoreFailsGracefully`**: null cursor + hasMore=true -> existing guard breaks loop gracefully

## Non-goals

- No changes to personalRecordIds parity semantics (LWW behavior)
- No cursor persistence across pull invocations (separate enhancement)
- No portal/edge function changes (already supports unlimited pages)
- No UI changes (existing `SyncingWithProgress` already renders per-page progress)

### CODE REVIEW

File: shared/src/commonTest/kotlin/com/devil/phoenixproject/data/sync/PortalPullPaginationTest.kt [hunk 1]
```diff
@@ -643,6 +643,144 @@ class PortalPullPaginationTest {
...
+                    hasMore = true,
+                    nextCursor = "cursor-loop", // repeated!
+                    routines = listOf(PullRoutineDto(id = "r2", name = "R2")),
+                ),
+            ),
+        )
+
+        val result = createManager().sync()
+
+        assertTrue(result.isFailure, "Repeated cursor must fail the pull")
```
[the remark this answers is not in the document]
Comment (9thLevelSoftware) [reply shared/src/commonTest/kotlin/com/devil/phoenixproject/data/sync/PortalPullPaginationTest.kt]: Fixed in commit e701427: Changed `pullWithRepeatedCursorFailsWithDiagnosticError` and `pullWithBlankCursorFailsWithDiagnosticError` to assert `SyncState.PartialSuccess` with `pullError` instead of `result.isFailure`, since `sync()` returns `Result.success(syncTimeEpoch)` when push succeeds.

File: shared/src/commonTest/kotlin/com/devil/phoenixproject/data/sync/PortalPullPaginationTest.kt [hunk 2]
```diff
@@ -643,6 +643,144 @@ class PortalPullPaginationTest {
...
+                    syncTime = 1_740_000_000_000L + index,
+                    hasMore = !isLast,
+                    nextCursor = if (isLast) null else "cursor-page-${index + 1}",
+                    routines = listOf(PullRoutineDto(id = "r$index", name = "Routine $index")),
+                ),
+            )
+        }
+        fakeApi.pullResultsQueue = pages
+
+        val result = createManager().sync()
```
[the remark this answers is not in the document]
Comment (9thLevelSoftware) [reply shared/src/commonTest/kotlin/com/devil/phoenixproject/data/sync/PortalPullPaginationTest.kt]: Fixed in commit e701427: Injected virtual-clock `ClientRateLimiter(nowMs={limiterClockMs}, waitFor={delayMs->limiterClockMs+=delayMs})` matching the pattern from `multiPagePullConsumesLimiterTokenPerPage`. Also wired `fakeApi.pullTimestampSourceMs` so all 101 pages complete deterministically.
